sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -106,6 +107,22 @@ object TypeCoercion {
     case (t1, t2) => findTypeForComplex(t1, t2, findTightestCommonType)
   }
 
+  /**
+   * Finds a wider decimal type between the two supplied decimal types without
+   * any loss of precision.
+   */
+  def findWiderDecimalType(d1: DecimalType, d2: DecimalType): Option[DecimalType] = {
+    val scale = max(d1.scale, d2.scale)
+    val range = max(d1.precision - d1.scale, d2.precision - d2.scale)
+
+    // Check the resultant decimal type does not exceed the allowable limits.
+    if (range + scale <= DecimalType.MAX_PRECISION && scale <= DecimalType.MAX_SCALE) {
```

Comment:
We need `scale <= DecimalType.MAX_SCALE`? `DecimalType.scale` has been already validated?

Reply:
@maropu OK.. i will remove this check.